setup.py

```diff
@@ -1,8 +1,22 @@
 """Setup script for lasio"""
 
+import lasio
+import re
 from setuptools import setup
 import os
 
+__init__file = os.path.join('lasio', '__init__.py')
+with open(__init__file, 'r') as f:
+    file_contents = f.read()
+
+changed_file_contents = re.sub(r"(__version__\s?=\s?)[\"\'\w()]+", fr"""\g<1>'{lasio.version()}'""", file_contents)
+
+if changed_file_contents == file_contents:
+    raise Exception("Failed to replace version call with hardcoded value. See LF-46097")
```

Comment:
What if the file contains exactly the string you are replacing? That doesn't seem like a failure.

I suppose you can argue that this fix should be removed in that instance...

Reply:
In the future, if they changed it to `__version__ = 0.40` like they did back in `__version__ = 0.23` my search will still match because I included ' and " in the character set. However, this might replace it with itself and raise the exception when it's fine, like you say 🤔

Reply:
Actually the current regex doesn't match 'x.y.z', since there is no dot character in the character group after the equals. In that situation, you get a broken string after that, but it will not fail this check. So that need fixing.

Reply:
Updated